src/renderers/shaders/ShaderChunk/lights_pars_begin.glsl.js

```diff
@@ -144,9 +152,11 @@ float getSpotAttenuation( const in float coneCosine, const in float penumbraCosi
 		float decay;
 		float coneCos;
 		float penumbraCos;
+		bool hasIESProfile;
 	};
 
 	uniform SpotLight spotLights[ NUM_SPOT_LIGHTS ];
+	uniform sampler2D iesProfiles[ NUM_SPOT_LIGHTS ];
```

Comment:
My only concern here is that with many spotlights you could quickly eat away at the max number of texture samplers even if no ies profiles are used on the spotlights. The texture array approach we used in gpu-pathtracer is the really generalized approach to this issue but I'm wondering if a simpler solution is to provide a minimal set of ies textures to the shader and use an index in the SpotLight struct? No ies profile would be indicated by - 1.

I'm not sure if this would work in WebGL 1, though, since I don't think it supports dynamically accessed arrays...

```glsl
	struct SpotLight {
		// ...
		float penumbraCos;
		int iesProfile;
	};
	uniform SpotLight spotLights[ NUM_SPOT_LIGHTS ];
	uniform sampler2D iesProfiles[ NUM_IES_PROFILES ];
```

Reply:
I tried this. An issue I had is that I could not use iesProfiles[ spotLight.iesProfile ] as sampler arrays must be looked up with constant indices only, fine for the unrolled loop case.

Reply:
Packing profiles into a single texture is a good approach, and if we stick with 1D profiles can just use `iesProfile` as one of the dimensions. Happy to go with this if you agree.

Reply:
So we would have one main `sampler2D iesProfiles;` and just dynamically update that as new profiles are requested.

Reply:
A concatenated texture would sound good to me and work in WebGL1 for 1D IES profiles as long we make sure that it doesn't get marked as needing reupload every render.

I'd wait for @mrdoob and @Mugen87 thoughts before updating the implementation, though.

Reply:
Indeed, a `needsUpdate` only when the profile set is changed was what I was thinking.

Reply:
Showing multiple 1D IES profiles that are combined into a single texture.

Now there's a single `uniform sampler2D iesProfiles;` and each light indexes into single lines of this.

![Screenshot from 2022-09-06 20-46-28](https://user-images.githubusercontent.com/868396/188725215-acbebd82-0015-4814-8c96-00bd1161d1d4.png)

Just needs a bit of tidy and then I'll submit for further review.

Reply:
Slightly better screenshot of the above, easier to see the contribution of each light.

![Screenshot from 2022-09-07 08-45-57](https://user-images.githubusercontent.com/868396/188820101-478af3b0-ce3e-4ac4-976b-ff5501754b04.png)